src/DistributedArrays.jl

```diff
@@ -880,13 +880,37 @@ for f in (:+, :-, :div, :mod, :rem, :&, :|, :$)
             B = samedist(A, B)
             map_localparts($f, A, B)
         end
+        function ($f){T}(A::DArray{T}, B::Array{T})
+            size(A) == size(B) || throw(DimensionMismatch("first array has size $(size(A)) but second array has $(size(B))"))
+            return DArray(A) do I
+                ($f)(localpart(A), B[I...])
```

Comment:
Wouldn't it be better to distribute B first to the processes? This way all of B will be serialised as part of the closure.